storagetransfer/storagetransfer_test.go

```diff
@@ -188,6 +189,120 @@ func TestGetLatestTransferOperation(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	buf := new(bytes.Buffer)
+
+	rootDirectory, err := ioutil.TempDir("", "download-to-posix-test")
+	if err != nil {
+		t.Errorf("download_to_posix: %#v", err)
+	}
+	defer os.RemoveAll(rootDirectory)
+
+	sinkAgentPoolName := "" //use default agent pool
+	gcsSourcePath := rootDirectory + "/"
+
+	resp, err := downloadToPosix(buf, tc.ProjectID, sinkAgentPoolName, gcsSinkBucket, gcsSourcePath, rootDirectory)
+	defer cleanupSTSJob(resp.Name, tc.ProjectID)
+
+	if err != nil {
+		t.Errorf("download_to_posix: %#v", err)
+	}
+
+	got := buf.String()
+	if want := "transferJobs/"; !strings.Contains(got, want) {
+		t.Errorf("download_to_posix: got %q, want %q", got, want)
+	}
+}
+
+func TestTransferFromPosix(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	buf := new(bytes.Buffer)
+
+	rootDirectory, err := ioutil.TempDir("", "transfer-from-posix-test")
+	if err != nil {
+		t.Errorf("transfer_from_posix: %#v", err)
+	}
+	defer os.RemoveAll(rootDirectory)
+
+	sourceAgentPoolName := "" //use default agent pool
+
+	resp, err := transferFromPosix(buf, tc.ProjectID, sourceAgentPoolName, rootDirectory, gcsSinkBucket)
+	defer cleanupSTSJob(resp.Name, tc.ProjectID)
+
+	if err != nil {
+		t.Errorf("transfer_from_posix: %#v", err)
+	}
+
+	got := buf.String()
+	if want := "transferJobs/"; !strings.Contains(got, want) {
+		t.Errorf("transfer_from_posix: got %q, want %q", got, want)
+	}
+}
+
+func TestTransferBetweenPosix(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	buf := new(bytes.Buffer)
+
+	rootDirectory, err := ioutil.TempDir("", "transfer-between-posix-test-source")
+	if err != nil {
+		t.Errorf("transfer_between_posix: %#v", err)
+	}
+	defer os.RemoveAll(rootDirectory)
+
+	destinationDirectory, err := ioutil.TempDir("", "transfer-between-posix-test-sink")
+	if err != nil {
+		t.Errorf("transfer_between_posix: %#v", err)
+	}
+	defer os.RemoveAll(destinationDirectory)
+
+	sourceAgentPoolName := "" //use default agent pool
+	sinkAgentPoolName := ""   //use default agent pool
+
+	resp, err := transferBetweenPosix(buf, tc.ProjectID, sourceAgentPoolName, sinkAgentPoolName, rootDirectory, destinationDirectory, gcsSinkBucket)
+	if err != nil {
+		t.Errorf("transfer_between_posix: %#v", err)
+	}
+	defer cleanupSTSJob(resp.Name, tc.ProjectID)
+
+	got := buf.String()
+	if want := "transferJobs/"; !strings.Contains(got, want) {
+		t.Errorf("transfer_between_posix: got %q, want %q", got, want)
+	}
+}
+
+func TestTransferUsingManifest(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	buf := new(bytes.Buffer)
+
+	rootDirectory, err := ioutil.TempDir("", "transfer-using-manifest-test")
+	if err != nil {
+		t.Errorf("transfer_using_manifest: %#v", err)
+	}
+	defer os.RemoveAll(rootDirectory)
+
+	sourceAgentPoolName := "" //use default agent pool
+	object := sc.Bucket(gcsSourceBucket).Object("manifest.csv")
+	defer object.Delete(context.Background())
+	manifestLocation := "gs://" + gcsSourceBucket + "/manifest.csv"
```

Comment:
Maybe better to pass gcsSourceBucket as an arg and construct this string in-line in the sample? Seems important to be clear that users must use the gs:// prefix.